test/distributed/test_c10d.py

```diff
@@ -3054,6 +3054,24 @@ def test_nccl_errors_blocking_sigkill(self):
     def test_nccl_errors_blocking_sigterm(self):
         self._test_nccl_errors_blocking(lambda : os.kill(os.getpid(), signal.SIGTERM))
 
+    @requires_nccl()
+    @requires_nccl_version(2400, "Need NCCL 2.4+ for error checking")
+    @skip_if_lt_x_gpu(3)
+    def test_nccl_blocking_wait_with_barrier(self):
+        os.environ["NCCL_BLOCKING_WAIT"] = "1"
+        store = c10d.FileStore(self.file_name, self.world_size)
+        process_group = c10d.ProcessGroupNCCL(
+            store,
+            self.rank,
+            self.world_size,
+            timeout=timedelta(seconds=self.op_timeout_sec))
+        process_group.barrier().wait()
```

Comment:
Why do we need this extra barrier call for all ranks? This test should work as expected even if there were no collectives operations before, correct?

Reply:
The nccl communicator is initialized (ncclCommInitRank) during the first collective call. NCCL unfortunately doesn't provide any API to timeout the communicator initialization if something fails. As a result, if we have only one barrier call and its run on a single rank, it will be stuck forerver.